src/Service.Tests/Unittests/SqlQueryExecutorUnitTests.cs

```diff
@@ -318,6 +318,62 @@ Mock<MsSqlQueryExecutor> queryExecutor
             Assert.AreEqual(3, queryExecutorLogger.Invocations.Count);
         }
 
+        /// <summary>
+        /// Validates that a query successfully executes within two retries by checking that the SqlQueryExecutor logger
+        /// was invoked the expected number of times.
+        /// </summary>
+        [DataTestMethod, TestCategory(TestCategory.MSSQL)]
+        [DataRow(false, 4, DisplayName = "Data should be successfully read when amount of data being read is less than max-db-response-size-mb")]
+        [DataRow(false, 5, DisplayName = "Data should be successfully read when amount of data being read is less than max-db-response-size-mb")]
+        [DataRow(true, 6, DisplayName = "Exception should be thrown when amount of data read is more than max-db-response-size-mb")]
```

Comment:
why is magic number `6` input result in a value more than `max-db-response-size-mb`?

Reply:
This was explained in lines 357-359. Ive moved it to the summary description. hopefully should be clearer.